[5/n][torch/elastic][upstream] Move torchelastic/agent to torch/distributed/elastic/agent

aivanou opened the pull request:
Summary: Move torchelastic/agent to torch/distributed/elastic/agent

Test Plan:
buck test mode/dev-nosan //pytorch/elastic/torchelastic/...
      buck test mode/dev-nosan //caffe2/test/distributed/elastic/agent/server/test/...

Differential Revision: D27173271

pritamdamania87 commented on test/distributed/elastic/agent/server/test/__init__.py:
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env python3
```
These files are already under the top level test/ directory, do we need another test/ suffix at the end?

pritamdamania87 commented on test/distributed/elastic/agent/server/test/api_test.py:
```diff
@@ -0,0 +1,556 @@
+#!/usr/bin/env python3
```
Don't we need to add these files to `run_test.py` so that the tests are actually run in CI?

aivanou committed:
[5/n][torch/elastic][upstream] Move torchelastic/agent to torch/distributed/elastic/agent (#54343)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/54343

Move torchelastic/agent to torch/distributed/elastic/agent

Test Plan:
buck test mode/dev-nosan //pytorch/elastic/torchelastic/...
      buck test mode/dev-nosan //caffe2/test/distributed/elastic/agent/server/test/...

Reviewed By: kiukchung, wilson100hong

Differential Revision: D27173271

fbshipit-source-id: ccbc0fbd32c0fab946116830a59a4b5d80d3abdc
